engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -1523,19 +1544,33 @@ public LogicalPlan getLogicalPlan()
    */
   public List<PTOperator> getOperators(OperatorMeta logicalOperator)
   {
+    /* During dynamic plan change, operators are added in logical plan first and then
+       physical plan is changed. There is a race, when REST api try to get information
+       about an operator which is not yet added in physical plan causes
+       NullPointerException here. null return value is handled at caller appropriately.
+     */
+    if (this.logicalToPTOperator.get(logicalOperator) == null) {
+      return null;
+    }
     return this.logicalToPTOperator.get(logicalOperator).partitions;
   }
 
   public Collection<PTOperator> getAllOperators(OperatorMeta logicalOperator)
   {
+    if (this.logicalToPTOperator.get(logicalOperator) == null) {
+      return null;
+    }
     return this.logicalToPTOperator.get(logicalOperator).getAllOperators();
   }
 
   public List<PTOperator> getLeafOperators()
   {
     List<PTOperator> operators = new ArrayList<>();
     for (OperatorMeta opMeta : dag.getLeafOperators()) {
-      operators.addAll(getAllOperators(opMeta));
+      Collection<PTOperator> allOpers = getAllOperators(opMeta);
```

Comment:
getAllOperators can return null when dag change in progress as logical plan is changed first and then physical plan is changed. This check is added to prevent null pointer exception during rest api calls when api tries to get physical operators for a logical operator but physical plan is not changed yet.